internal/distributed/proxy/service.go

```diff
@@ -139,9 +144,15 @@ func (s *Server) startHTTPServer(port int) {
 	}
 }
 
-func (s *Server) startGrpcLoop(grpcPort int) {
-	defer s.wg.Done()
+func (s *Server) startGrpcLoop(grpcPort, grpcInternalPort int) {
```

Comment:
Do not use goroutine to start grpc and internal grpc within this method, i saw 336 lines already use goroutine.